Sources/ParseSwift/API/URLSession+async.swift

```diff
@@ -0,0 +1,89 @@
+//
+//  URLSession+async.swift
+//  ParseSwift
+//
+//  Created by Corey Baker on 7/22/20.
+//  Copyright © 2020 Parse. All rights reserved.
+//
+
+import Foundation
+import Combine
+
+extension URLSession {
+
+    internal func asyncDataTask(with request: URLRequest,
+                                completion: @escaping(Result<Data, ParseError>) -> Void) {
+
+        let semaphore = DispatchSemaphore(value: 0)
+        dataTask(with: request) { (responseData, urlResponse, responseError) in
+
+            guard let responseData = responseData else {
```

Comment:
Basically nothing inside the completion fires without the semaphore because it gets killed when it goes out-of-scope.

Reply:
Right, because nothing is hold on to `dataTask`. And why do you want to remove the semaphore?

Reply:
It’s blocking the thread with the semaphore there, so it isn’t truly Async. Leaving the semaphore there is the same as the sync case